spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/servicebus/factory/DefaultServiceBusTopicClientFactory.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.servicebus.factory;
+
+import com.microsoft.azure.management.servicebus.Topic;
+import com.microsoft.azure.servicebus.*;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import com.microsoft.azure.spring.cloud.context.core.AzureAdmin;
+import com.microsoft.azure.spring.cloud.context.core.Tuple;
+import com.microsoft.azure.spring.integration.core.Memoizer;
+import com.microsoft.azure.spring.integration.servicebus.ServiceBusRuntimeException;
+import org.springframework.util.Assert;
+
+import java.util.function.Function;
+
+/**
+ * Default implementation of {@link ServiceBusTopicClientFactory}.
+ * Client will be cached to improve performance
+ *
+ * @author Warren Zhu
+ */
+public class DefaultServiceBusTopicClientFactory extends AbstractServiceBusSenderFactory
+        implements ServiceBusTopicClientFactory {
```

Comment:
remove implements ServiceBusTopicClientFactory since AbstractServiceBusSenderFactory already implements it?

Reply:
We need it here since `AbstractServiceBusSenderFactory` only provide partial implementation of `ServiceBusTopicClientFactory`